safe-global/safe-wallet-web
# [Counterfactual] Add deploy safe flow

## What it solves

Resolves #3195 

## How this PR fixes it

- Adds a new flow for deploying a safe
- Adds a new section on the settings page that is only visible for counterfactual safes

## How to test it

1. Create a counterfactual safe
2. Navigate to the settings page
3. Observe a new section for deploying the safe
4. Press on Activate now
5. Observe a new safe deployment flow
6. Execute the transaction
7. Observe the settings section is not visible anymore
8. Observe the safe is deployed on-chain

## Screenshots
<img width="1274" alt="Screenshot 2024-02-05 at 10 00 11" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/aceb51cd-5ca0-4dfb-9de0-b02f08ae6743">
<img width="722" alt="Screenshot 2024-02-05 at 10 00 24" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/5b6af6b9-1b05-4f50-9504-89a73c27e024">

## Checklist
